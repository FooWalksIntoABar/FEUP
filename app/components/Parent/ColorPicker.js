import React, { Component } from 'react'
import { Picker, Item } from 'native-base'
import PropTypes from 'prop-types'

export class ColorPicker extends Component {
  onValueChange = value => {
    this.props.onColorChange(this.props.colors[value].code)
  }

  render () {
    const pickerItems = this.props.colors.map((color, index) => (<Picker.Item label={color.name} value={index} key={index} />))
    return (
      <Item picker style={{ borderColor: this.props.color }}>
        <Picker
          mode='dropdown'
          style={{ width: 120 }}
          selectedValue={this.props.colors.findIndex(color => color.code === this.props.color)}
          onValueChange={this.onValueChange}
        >
          {pickerItems}
        </Picker>
      </Item>
    )
  }
}

ColorPicker.propTypes = {
  color: PropTypes.string.isRequired,
  colors: PropTypes.array.isRequired,
  onColorChange: PropTypes.func.isRequired
}
