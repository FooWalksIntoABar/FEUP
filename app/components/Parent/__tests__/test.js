import 'react-native'
import React from 'react'
import { Button } from 'native-base'
import { configure, shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import Adapter from 'enzyme-adapter-react-16'

import { BottomButton } from '../BottomButton'
import { ColorPicker } from '../ColorPicker'
import { DurationPickers } from '../DurationPickers'
import { ImagePickerButtons } from '../ImagePickerButtons'
import { InvalidateList } from '../InvalidateList'
import { ItemPicker } from '../ItemPicker'
import { PeriodicityPicker } from '../PeriodicityPicker'
import { PhotoPickerButton } from '../PhotoPickerButton'
import { RoutineItem } from '../RoutineItem'
import { SelectChildPicker } from '../SelectChildPicker'
import { SortableList } from '../SortableList'
import { SortableListItem } from '../SortableListItem'
import { SortableListItemThumbnail } from '../SortableListItemThumbnail'

import { activities } from '../../../constants/mockTestData'

configure({ adapter: new Adapter() })

describe('Slide components', () => {
  it('renders BottomButton correctly', () => {
    const wrapper = shallow(<BottomButton
      color={''}
      text={''}
      onPress={jest.fn()} />)
    expect(toJson(wrapper)).toMatchSnapshot()
    wrapper.unmount()
  })

  it('renders ColorPicker correctly 1', () => {
    const wrapper = shallow(<ColorPicker
      color={1}
      colors={[]}
      onColorChange={jest.fn()} />)
    expect(toJson(wrapper)).toMatchSnapshot()
    wrapper.unmount()
  })

  it('renders ColorPicker correctly 2', () => {
    const wrapper = shallow(<ColorPicker
      color={0}
      colors={[1, 2]}
      onColorChange={jest.fn()} />)
    wrapper.instance().onValueChange(1)
    wrapper.instance().onValueChange(999)
    wrapper.instance().getSelectedValue()
    wrapper.unmount()
  })

  it('renders DurationPickers correctly', () => {
    const wrapper = shallow(<DurationPickers
      onDurationChange={jest.fn()}
      time={{}}
      color={''} />)
    expect(toJson(wrapper)).toMatchSnapshot()
    wrapper.instance().toggleDetailedView()
    wrapper.instance().changeGoalValue({ replace: jest.fn() })
    wrapper.instance().changeMaxValue({ replace: jest.fn() })
    wrapper.instance().changeMinValue({ replace: jest.fn() })
    wrapper.unmount()
  })

  it('renders ImagePickerButtons correctly', async () => {
    const wrapper = shallow(<ImagePickerButtons
      color={''}
      onPhotoChange={jest.fn()}
      onImageChange={jest.fn()}
      photo={''}
      image={''} />)
    expect(toJson(wrapper)).toMatchSnapshot()
    wrapper.setState({ photoView: true })
    wrapper.find(Button).at(0).props().onPress()
    wrapper.find(Button).at(1).props().onPress()
    wrapper.instance().togglePhotoView()
    wrapper.instance().onImageChanged(0)
    wrapper.setProps({ image: 'drawers' })
    wrapper.instance().getSelectedImageName()
    wrapper.unmount()
  })

  it('renders InvalidateList correctly', () => {
    const wrapper = shallow(<InvalidateList
      activities={activities}
      onInvalidatePress={jest.fn()} />)
    expect(toJson(wrapper)).toMatchSnapshot()
    wrapper.find(Button).at(0).props().onPress()
    wrapper.instance().formatTime(-10)
    wrapper.unmount()
  })

  it('renders ItemPicker correctly', () => {
    const wrapper = shallow(<ItemPicker
      items={[ { label: 'label' }, { label: 'other' } ]}
      selected={{}}
      onValueChange={jest.fn()} />)
    expect(toJson(wrapper)).toMatchSnapshot()
    wrapper.unmount()
  })

  it('renders PeriodicityPicker correctly', () => {
    const wrapper = shallow(<PeriodicityPicker
      isRepeat
      color={''}
      periodicity={[]}
      togglePeriodicity={jest.fn()}
      toggleIsRepeat={jest.fn()} />)
    expect(toJson(wrapper)).toMatchSnapshot()
    wrapper.find(Button).at(0).props().onPress()
    wrapper.unmount()
  })

  it('renders PhotoPickerButton correctly', async () => {
    const wrapper = shallow(<PhotoPickerButton
      color={''}
      photo={{}}
      onPhotoChange={jest.fn()} />)
    expect(toJson(wrapper)).toMatchSnapshot()
    wrapper.find(Button).at(0).props().onPress()
    wrapper.find(Button).at(1).props().onPress()
    wrapper.unmount()
  })

  it('renders RoutineItem correctly', () => {
    const wrapper = shallow(<RoutineItem
      image={0}
      title={''} />)
    expect(toJson(wrapper)).toMatchSnapshot()
    wrapper.unmount()
  })

  it('renders SelectChildPicker correctly', () => {
    const wrapper = shallow(<SelectChildPicker
      children={[{ image: '' }]}
      selected={0}
      onChildChanged={jest.fn()} />)
    expect(toJson(wrapper)).toMatchSnapshot()
    wrapper.unmount()
  })

  it('renders SortableList correctly', () => {
    const wrapper = shallow(<SortableList
      items={[{}, {}, {}]}
      color={''}
      onItemPress={jest.fn()} />)
    expect(toJson(wrapper)).toMatchSnapshot()
    wrapper.unmount()
  })

  it('renders SortableListItem correctly', () => {
    const wrapper = shallow(<SortableListItem
      item={{}}
      index={1}
      moveItemUp={jest.fn()}
      color={''}
      onItemPress={jest.fn()} />)
    expect(toJson(wrapper)).toMatchSnapshot()
    wrapper.find(Button).at(0).props().onPress()
    wrapper.instance().onItemPress()
    wrapper.unmount()
  })

  it('renders SortableListItemThumbnail correctly', () => {
    const wrapper = shallow(<SortableListItemThumbnail
      color={''}
      source={{}}
      isPhoto />)
    expect(toJson(wrapper)).toMatchSnapshot()
    wrapper.unmount()
  })
})
