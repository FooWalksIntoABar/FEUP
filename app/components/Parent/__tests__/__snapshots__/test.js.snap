// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Slide components renders BottomButton correctly 1`] = `
<Component
  style={
    Object {
      "padding": 15,
      "width": "100%",
    }
  }
>
  <Styled(Button)
    onPress={[MockFunction]}
    style={
      Object {
        "backgroundColor": "",
        "justifyContent": "center",
        "width": "100%",
      }
    }
  >
    <Styled(Text) />
  </Styled(Button)>
</Component>
`;

exports[`Slide components renders ColorPicker correctly 1 1`] = `
<Styled(Item)
  picker={true}
  style={
    Object {
      "borderColor": 1,
    }
  }
>
  <Styled(PickerNB)
    mode="dropdown"
    onValueChange={[Function]}
    selectedValue={-1}
    style={
      Object {
        "width": 120,
      }
    }
  >
    <Item
      key="999"
      label="Outra"
      value={999}
    />
  </Styled(PickerNB)>
</Styled(Item)>
`;

exports[`Slide components renders DurationPickers correctly 1`] = `
<Component>
  <Styled(Item)
    stackedLabel={true}
  >
    <Styled(Label)>
      Duração (minutos)
    </Styled(Label)>
    <Styled(Input)
      keyboardType="numeric"
      onChangeText={[Function]}
      textContentType="telephoneNumber"
    />
  </Styled(Item)>
  <Component
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "flex-end",
      }
    }
  >
    <Styled(Button)
      onPress={[Function]}
      transparent={true}
    >
      <Styled(Text)
        style={
          Object {
            "color": "",
          }
        }
      >
        Mais opções de duração
      </Styled(Text)>
    </Styled(Button)>
  </Component>
</Component>
`;

exports[`Slide components renders ImagePickerButtons correctly 1`] = `
<Component>
  <Styled(Item)
    stackedLabel={true}
  >
    <Styled(Label)>
      Imagem
    </Styled(Label)>
    <Component
      style={
        Object {
          "flexDirection": "column",
          "width": "100%",
        }
      }
    >
      <Styled(Item)
        style={
          Object {
            "borderColor": "transparent",
          }
        }
      >
        <ItemPicker
          items={
            Array [
              "Armário",
              "Blocos",
              "Cama",
              "Cereais",
              "Chuveiro",
              "Chá",
              "Dormir",
              "Escova de dentes",
              "Livro",
              "Lua",
              "Meias",
              "Mochila",
              "Pente",
              "Sapatilhas",
              "Sol",
              "T-Shirt",
              "Talheres",
              "Trabalhos",
            ]
          }
          onValueChange={[Function]}
        />
      </Styled(Item)>
    </Component>
  </Styled(Item)>
  <Component
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "flex-end",
      }
    }
  >
    <Styled(Button)
      onPress={[Function]}
      transparent={true}
    >
      <Styled(Text)
        style={
          Object {
            "color": "",
          }
        }
      >
        Escolher fotografia
      </Styled(Text)>
    </Styled(Button)>
  </Component>
</Component>
`;

exports[`Slide components renders InvalidateList correctly 1`] = `
<Styled(List)
  style={
    Object {
      "width": "100%",
    }
  }
>
  <Styled(ListItem)
    button={true}
    key="0"
  >
    <Styled(Left)>
      <SortableListItemThumbnail
        color="#7d84b2"
        isPhoto={true}
        source={
          Object {
            "uri": "bedroom",
          }
        }
      />
      <Component
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <Styled(Text)
          style={
            Object {
              "alignSelf": "flex-start",
            }
          }
        >
          Fazer a cama
        </Styled(Text)>
        <Component
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
            }
          }
        >
          <Image
            source={1}
            style={
              Array [
                Object {
                  "height": 12,
                  "margin": 2,
                  "width": 12,
                },
                false,
              ]
            }
          />
          <Image
            source={1}
            style={
              Array [
                Object {
                  "height": 12,
                  "margin": 2,
                  "width": 12,
                },
                false,
              ]
            }
          />
          <Image
            source={1}
            style={
              Array [
                Object {
                  "height": 12,
                  "margin": 2,
                  "width": 12,
                },
                false,
              ]
            }
          />
          <Styled(Text)
            note={true}
            numberOfLines={1}
            style={
              Object {
                "alignSelf": "flex-start",
              }
            }
          >
            - Falhada em NaN:NaN
          </Styled(Text)>
        </Component>
      </Component>
    </Styled(Left)>
    <Styled(Right)>
      <Styled(Button)
        onPress={[Function]}
        rounded={true}
        style={
          Object {
            "backgroundColor": "#e63b2e",
          }
        }
      >
        <Styled(Icon)
          name="md-close"
        />
      </Styled(Button)>
    </Styled(Right)>
  </Styled(ListItem)>
  <Styled(ListItem)
    button={true}
    key="1"
  >
    <Styled(Left)>
      <SortableListItemThumbnail
        color="#0e79b2"
        isPhoto={true}
        source={
          Object {
            "uri": undefined,
          }
        }
      />
      <Component
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <Styled(Text)
          style={
            Object {
              "alignSelf": "flex-start",
            }
          }
        >
          Lavar os dentes
        </Styled(Text)>
        <Component
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
            }
          }
        >
          <Image
            source={1}
            style={
              Array [
                Object {
                  "height": 12,
                  "margin": 2,
                  "width": 12,
                },
                false,
              ]
            }
          />
          <Image
            source={1}
            style={
              Array [
                Object {
                  "height": 12,
                  "margin": 2,
                  "width": 12,
                },
                false,
              ]
            }
          />
          <Image
            source={1}
            style={
              Array [
                Object {
                  "height": 12,
                  "margin": 2,
                  "width": 12,
                },
                false,
              ]
            }
          />
          <Styled(Text)
            note={true}
            numberOfLines={1}
            style={
              Object {
                "alignSelf": "flex-start",
              }
            }
          >
            - Falhada em NaN:NaN
          </Styled(Text)>
        </Component>
      </Component>
    </Styled(Left)>
    <Styled(Right)>
      <Styled(Button)
        onPress={[Function]}
        rounded={true}
        style={
          Object {
            "backgroundColor": "#e63b2e",
          }
        }
      >
        <Styled(Icon)
          name="md-close"
        />
      </Styled(Button)>
    </Styled(Right)>
  </Styled(ListItem)>
  <Styled(ListItem)
    button={true}
    key="2"
  >
    <Styled(Left)>
      <SortableListItemThumbnail
        color="#7fb800"
        isPhoto={true}
        source={
          Object {
            "uri": undefined,
          }
        }
      />
      <Component
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <Styled(Text)
          style={
            Object {
              "alignSelf": "flex-start",
            }
          }
        >
          Vestir
        </Styled(Text)>
        <Component
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
            }
          }
        >
          <Image
            source={1}
            style={
              Array [
                Object {
                  "height": 12,
                  "margin": 2,
                  "width": 12,
                },
                false,
              ]
            }
          />
          <Image
            source={1}
            style={
              Array [
                Object {
                  "height": 12,
                  "margin": 2,
                  "width": 12,
                },
                false,
              ]
            }
          />
          <Image
            source={1}
            style={
              Array [
                Object {
                  "height": 12,
                  "margin": 2,
                  "width": 12,
                },
                false,
              ]
            }
          />
          <Styled(Text)
            note={true}
            numberOfLines={1}
            style={
              Object {
                "alignSelf": "flex-start",
              }
            }
          >
            - Falhada em NaN:NaN
          </Styled(Text)>
        </Component>
      </Component>
    </Styled(Left)>
    <Styled(Right)>
      <Styled(Button)
        onPress={[Function]}
        rounded={true}
        style={
          Object {
            "backgroundColor": "#e63b2e",
          }
        }
      >
        <Styled(Icon)
          name="md-close"
        />
      </Styled(Button)>
    </Styled(Right)>
  </Styled(ListItem)>
  <Styled(ListItem)
    button={true}
    key="3"
  >
    <Styled(Left)>
      <SortableListItemThumbnail
        color="#37c1f0"
        isPhoto={true}
        source={
          Object {
            "uri": undefined,
          }
        }
      />
      <Component
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <Styled(Text)
          style={
            Object {
              "alignSelf": "flex-start",
            }
          }
        >
          Tomar banho
        </Styled(Text)>
        <Component
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
            }
          }
        >
          <Image
            source={1}
            style={
              Array [
                Object {
                  "height": 12,
                  "margin": 2,
                  "width": 12,
                },
                false,
              ]
            }
          />
          <Image
            source={1}
            style={
              Array [
                Object {
                  "height": 12,
                  "margin": 2,
                  "width": 12,
                },
                false,
              ]
            }
          />
          <Image
            source={1}
            style={
              Array [
                Object {
                  "height": 12,
                  "margin": 2,
                  "width": 12,
                },
                false,
              ]
            }
          />
          <Styled(Text)
            note={true}
            numberOfLines={1}
            style={
              Object {
                "alignSelf": "flex-start",
              }
            }
          >
            - Falhada em NaN:NaN
          </Styled(Text)>
        </Component>
      </Component>
    </Styled(Left)>
    <Styled(Right)>
      <Styled(Button)
        onPress={[Function]}
        rounded={true}
        style={
          Object {
            "backgroundColor": "#e63b2e",
          }
        }
      >
        <Styled(Icon)
          name="md-close"
        />
      </Styled(Button)>
    </Styled(Right)>
  </Styled(ListItem)>
  <Styled(ListItem)
    button={true}
    key="4"
  >
    <Styled(Left)>
      <SortableListItemThumbnail
        color="#e43f6f"
        isPhoto={true}
        source={
          Object {
            "uri": undefined,
          }
        }
      />
      <Component
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <Styled(Text)
          style={
            Object {
              "alignSelf": "flex-start",
            }
          }
        >
          Preparar a mochila
        </Styled(Text)>
        <Component
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
            }
          }
        >
          <Image
            source={1}
            style={
              Array [
                Object {
                  "height": 12,
                  "margin": 2,
                  "width": 12,
                },
                false,
              ]
            }
          />
          <Image
            source={1}
            style={
              Array [
                Object {
                  "height": 12,
                  "margin": 2,
                  "width": 12,
                },
                false,
              ]
            }
          />
          <Image
            source={1}
            style={
              Array [
                Object {
                  "height": 12,
                  "margin": 2,
                  "width": 12,
                },
                false,
              ]
            }
          />
          <Styled(Text)
            note={true}
            numberOfLines={1}
            style={
              Object {
                "alignSelf": "flex-start",
              }
            }
          >
            - Falhada em NaN:NaN
          </Styled(Text)>
        </Component>
      </Component>
    </Styled(Left)>
    <Styled(Right)>
      <Styled(Button)
        onPress={[Function]}
        rounded={true}
        style={
          Object {
            "backgroundColor": "#e63b2e",
          }
        }
      >
        <Styled(Icon)
          name="md-close"
        />
      </Styled(Button)>
    </Styled(Right)>
  </Styled(ListItem)>
  <Styled(ListItem)
    button={true}
    key="5"
  >
    <Styled(Left)>
      <SortableListItemThumbnail
        color="#4bb3fd"
        isPhoto={true}
        source={
          Object {
            "uri": undefined,
          }
        }
      />
      <Component
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <Styled(Text)
          style={
            Object {
              "alignSelf": "flex-start",
            }
          }
        >
          Calçar os sapatos
        </Styled(Text)>
        <Component
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
            }
          }
        >
          <Image
            source={1}
            style={
              Array [
                Object {
                  "height": 12,
                  "margin": 2,
                  "width": 12,
                },
                false,
              ]
            }
          />
          <Image
            source={1}
            style={
              Array [
                Object {
                  "height": 12,
                  "margin": 2,
                  "width": 12,
                },
                false,
              ]
            }
          />
          <Image
            source={1}
            style={
              Array [
                Object {
                  "height": 12,
                  "margin": 2,
                  "width": 12,
                },
                false,
              ]
            }
          />
          <Styled(Text)
            note={true}
            numberOfLines={1}
            style={
              Object {
                "alignSelf": "flex-start",
              }
            }
          >
            - Falhada em NaN:NaN
          </Styled(Text)>
        </Component>
      </Component>
    </Styled(Left)>
    <Styled(Right)>
      <Styled(Button)
        onPress={[Function]}
        rounded={true}
        style={
          Object {
            "backgroundColor": "#e63b2e",
          }
        }
      >
        <Styled(Icon)
          name="md-close"
        />
      </Styled(Button)>
    </Styled(Right)>
  </Styled(ListItem)>
  <Styled(ListItem)
    button={true}
    key="6"
  >
    <Styled(Left)>
      <SortableListItemThumbnail
        color="#ff7f11"
        isPhoto={true}
        source={
          Object {
            "uri": undefined,
          }
        }
      />
      <Component
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <Styled(Text)
          style={
            Object {
              "alignSelf": "flex-start",
            }
          }
        >
          Tomar o pequeno almoço
        </Styled(Text)>
        <Component
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
            }
          }
        >
          <Image
            source={1}
            style={
              Array [
                Object {
                  "height": 12,
                  "margin": 2,
                  "width": 12,
                },
                false,
              ]
            }
          />
          <Image
            source={1}
            style={
              Array [
                Object {
                  "height": 12,
                  "margin": 2,
                  "width": 12,
                },
                false,
              ]
            }
          />
          <Image
            source={1}
            style={
              Array [
                Object {
                  "height": 12,
                  "margin": 2,
                  "width": 12,
                },
                false,
              ]
            }
          />
          <Styled(Text)
            note={true}
            numberOfLines={1}
            style={
              Object {
                "alignSelf": "flex-start",
              }
            }
          >
            - Falhada em NaN:NaN
          </Styled(Text)>
        </Component>
      </Component>
    </Styled(Left)>
    <Styled(Right)>
      <Styled(Button)
        onPress={[Function]}
        rounded={true}
        style={
          Object {
            "backgroundColor": "#e63b2e",
          }
        }
      >
        <Styled(Icon)
          name="md-close"
        />
      </Styled(Button)>
    </Styled(Right)>
  </Styled(ListItem)>
  <Styled(ListItem)
    button={true}
    key="7"
  >
    <Styled(Left)>
      <SortableListItemThumbnail
        color="#b0db43"
        isPhoto={true}
        source={
          Object {
            "uri": undefined,
          }
        }
      />
      <Component
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <Styled(Text)
          style={
            Object {
              "alignSelf": "flex-start",
            }
          }
        >
          Pentear cabelo
        </Styled(Text)>
        <Component
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
            }
          }
        >
          <Image
            source={1}
            style={
              Array [
                Object {
                  "height": 12,
                  "margin": 2,
                  "width": 12,
                },
                false,
              ]
            }
          />
          <Image
            source={1}
            style={
              Array [
                Object {
                  "height": 12,
                  "margin": 2,
                  "width": 12,
                },
                false,
              ]
            }
          />
          <Image
            source={1}
            style={
              Array [
                Object {
                  "height": 12,
                  "margin": 2,
                  "width": 12,
                },
                false,
              ]
            }
          />
          <Styled(Text)
            note={true}
            numberOfLines={1}
            style={
              Object {
                "alignSelf": "flex-start",
              }
            }
          >
            - Falhada em NaN:NaN
          </Styled(Text)>
        </Component>
      </Component>
    </Styled(Left)>
    <Styled(Right)>
      <Styled(Button)
        onPress={[Function]}
        rounded={true}
        style={
          Object {
            "backgroundColor": "#e63b2e",
          }
        }
      >
        <Styled(Icon)
          name="md-close"
        />
      </Styled(Button)>
    </Styled(Right)>
  </Styled(ListItem)>
</Styled(List)>
`;

exports[`Slide components renders ItemPicker correctly 1`] = `
<Styled(PickerNB)
  mode="dropdown"
  onValueChange={[MockFunction]}
  selectedValue={-1}
  style={
    Object {
      "width": "100%",
    }
  }
>
  <Item
    key="0"
    label={
      Object {
        "label": "label",
      }
    }
    value={0}
  />
  <Item
    key="1"
    label={
      Object {
        "label": "other",
      }
    }
    value={1}
  />
</Styled(PickerNB)>
`;

exports[`Slide components renders PeriodicityPicker correctly 1`] = `
<Component>
  <Styled(Item)
    stackedLabel={true}
  >
    <Styled(Label)>
      Periodicidade
    </Styled(Label)>
    <Component
      style={
        Object {
          "flexDirection": "row",
          "width": "100%",
        }
      }
    >
      <Styled(Button)
        key="0"
        onPress={[Function]}
        style={
          Object {
            "marginRight": 10,
            "paddingBottom": 0,
            "paddingTop": 10,
          }
        }
        transparent={true}
      >
        <Styled(Badge)
          style={
            Object {
              "backgroundColor": "transparent",
            }
          }
        >
          <Styled(Text)
            style={
              Object {
                "color": "#000",
              }
            }
          >
            S
          </Styled(Text)>
        </Styled(Badge)>
      </Styled(Button)>
      <Styled(Button)
        key="1"
        onPress={[Function]}
        style={
          Object {
            "marginRight": 10,
            "paddingBottom": 0,
            "paddingTop": 10,
          }
        }
        transparent={true}
      >
        <Styled(Badge)
          style={
            Object {
              "backgroundColor": "transparent",
            }
          }
        >
          <Styled(Text)
            style={
              Object {
                "color": "#000",
              }
            }
          >
            T
          </Styled(Text)>
        </Styled(Badge)>
      </Styled(Button)>
      <Styled(Button)
        key="2"
        onPress={[Function]}
        style={
          Object {
            "marginRight": 10,
            "paddingBottom": 0,
            "paddingTop": 10,
          }
        }
        transparent={true}
      >
        <Styled(Badge)
          style={
            Object {
              "backgroundColor": "transparent",
            }
          }
        >
          <Styled(Text)
            style={
              Object {
                "color": "#000",
              }
            }
          >
            Q
          </Styled(Text)>
        </Styled(Badge)>
      </Styled(Button)>
      <Styled(Button)
        key="3"
        onPress={[Function]}
        style={
          Object {
            "marginRight": 10,
            "paddingBottom": 0,
            "paddingTop": 10,
          }
        }
        transparent={true}
      >
        <Styled(Badge)
          style={
            Object {
              "backgroundColor": "transparent",
            }
          }
        >
          <Styled(Text)
            style={
              Object {
                "color": "#000",
              }
            }
          >
            Q
          </Styled(Text)>
        </Styled(Badge)>
      </Styled(Button)>
      <Styled(Button)
        key="4"
        onPress={[Function]}
        style={
          Object {
            "marginRight": 10,
            "paddingBottom": 0,
            "paddingTop": 10,
          }
        }
        transparent={true}
      >
        <Styled(Badge)
          style={
            Object {
              "backgroundColor": "transparent",
            }
          }
        >
          <Styled(Text)
            style={
              Object {
                "color": "#000",
              }
            }
          >
            S
          </Styled(Text)>
        </Styled(Badge)>
      </Styled(Button)>
      <Styled(Button)
        key="5"
        onPress={[Function]}
        style={
          Object {
            "marginRight": 10,
            "paddingBottom": 0,
            "paddingTop": 10,
          }
        }
        transparent={true}
      >
        <Styled(Badge)
          style={
            Object {
              "backgroundColor": "transparent",
            }
          }
        >
          <Styled(Text)
            style={
              Object {
                "color": "#000",
              }
            }
          >
            S
          </Styled(Text)>
        </Styled(Badge)>
      </Styled(Button)>
      <Styled(Button)
        key="6"
        onPress={[Function]}
        style={
          Object {
            "marginRight": 10,
            "paddingBottom": 0,
            "paddingTop": 10,
          }
        }
        transparent={true}
      >
        <Styled(Badge)
          style={
            Object {
              "backgroundColor": "transparent",
            }
          }
        >
          <Styled(Text)
            style={
              Object {
                "color": "#000",
              }
            }
          >
            D
          </Styled(Text)>
        </Styled(Badge)>
      </Styled(Button)>
    </Component>
    <Component
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginBottom": 15,
          "width": "100%",
        }
      }
    >
      <Styled(Text)>
        Repetir
      </Styled(Text)>
      <Styled(CheckBox)
        checked={true}
        color=""
        onPress={[MockFunction]}
        style={
          Object {
            "borderColor": "",
            "marginRight": 25,
          }
        }
      />
    </Component>
  </Styled(Item)>
</Component>
`;

exports[`Slide components renders PhotoPickerButton correctly 1`] = `
<Component>
  <Styled(Item)
    stackedLabel={true}
  >
    <Styled(Label)>
      Imagem
    </Styled(Label)>
    <Component
      style={
        Object {
          "flexDirection": "column",
          "width": "100%",
        }
      }
    >
      <Component
        style={
          Object {
            "alignItems": "center",
            "flexGrow": 1,
            "marginTop": 15,
            "paddingRight": 15,
          }
        }
      >
        <Image
          resizeMode="contain"
          source={Object {}}
          style={
            Object {
              "aspectRatio": 1,
              "width": "100%",
            }
          }
        />
      </Component>
      <Component
        style={
          Object {
            "flexDirection": "row",
            "justifyContent": "flex-start",
            "width": "100%",
          }
        }
      >
        <Styled(Button)
          onPress={[Function]}
          style={
            Array [
              Object {
                "flex": 1,
                "justifyContent": "center",
                "marginRight": 15,
                "marginVertical": 15,
              },
              Object {
                "backgroundColor": "",
              },
            ]
          }
        >
          <Styled(Text)>
            Camara
          </Styled(Text)>
        </Styled(Button)>
        <Styled(Button)
          onPress={[Function]}
          style={
            Array [
              Object {
                "flex": 1,
                "justifyContent": "center",
                "marginRight": 15,
                "marginVertical": 15,
              },
              Object {
                "backgroundColor": "",
              },
            ]
          }
        >
          <Styled(Text)>
            Galeria
          </Styled(Text)>
        </Styled(Button)>
      </Component>
    </Component>
  </Styled(Item)>
</Component>
`;

exports[`Slide components renders RoutineItem correctly 1`] = `
<Styled(ListItem)
  thumbnail={true}
>
  <Styled(Left)>
    <Styled(Thumbnail)
      source={0}
      square={true}
    />
  </Styled(Left)>
  <Styled(Body)>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    />
  </Styled(Body)>
  <Styled(Right)>
    <Styled(Button)
      transparent={true}
    >
      <Styled(Icon)
        name="md-arrow-forward"
      />
    </Styled(Button)>
  </Styled(Right)>
</Styled(ListItem)>
`;

exports[`Slide components renders SelectChildPicker correctly 1`] = `
<Component
  style={
    Object {
      "borderBottomColor": "lightgrey",
      "borderBottomWidth": 1,
      "width": "100%",
    }
  }
>
  <Component
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "width": "100%",
      }
    }
  >
    <Component
      style={
        Object {
          "alignItems": "center",
          "borderRadius": 35,
          "height": 70,
          "justifyContent": "center",
          "margin": 15,
          "overflow": "hidden",
          "width": 70,
        }
      }
    >
      <Image
        source={
          Object {
            "uri": "",
          }
        }
        style={
          Object {
            "height": "100%",
            "width": "100%",
          }
        }
      />
    </Component>
    <Styled(PickerNB)
      itemTextStyle={
        Object {
          "color": "red",
        }
      }
      mode="dropdown"
      onValueChange={[MockFunction]}
      selectedValue={0}
      style={
        Object {
          "flexGrow": 1,
          "height": "100%",
        }
      }
    >
      <Item
        key="0"
        value={0}
      />
    </Styled(PickerNB)>
  </Component>
  <Component
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "padding": 15,
        "paddingTop": 0,
        "width": "100%",
      }
    }
  >
    <Styled(Text)>
      Nível 
    </Styled(Text)>
    <Component
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
        }
      }
    >
      <Image
        source={1}
        style={
          Object {
            "height": 20,
            "marginRight": 5,
            "width": 20,
          }
        }
      />
      <Styled(Text) />
    </Component>
  </Component>
</Component>
`;

exports[`Slide components renders SortableList correctly 1`] = `
<Styled(List)
  style={
    Object {
      "width": "100%",
    }
  }
>
  <SortableListItem
    color=""
    index={0}
    item={Object {}}
    key="0"
    onItemPress={[MockFunction]}
  />
  <SortableListItem
    color=""
    index={1}
    item={Object {}}
    key="1"
    onItemPress={[MockFunction]}
  />
  <SortableListItem
    color=""
    index={2}
    item={Object {}}
    key="2"
    onItemPress={[MockFunction]}
  />
</Styled(List)>
`;

exports[`Slide components renders SortableListItem correctly 1`] = `
<Styled(ListItem)
  button={true}
  onPress={[Function]}
>
  <Styled(Left)>
    <SortableListItemThumbnail
      isPhoto={true}
      source={
        Object {
          "uri": undefined,
        }
      }
    />
    <Styled(Text) />
  </Styled(Left)>
  <Styled(Body) />
  <Styled(Right)>
    <Styled(Button)
      onPress={[Function]}
      style=""
    >
      <Styled(Icon)
        name="md-arrow-dropup"
      />
    </Styled(Button)>
  </Styled(Right)>
</Styled(ListItem)>
`;

exports[`Slide components renders SortableListItemThumbnail correctly 1`] = `
<Component
  style={
    Array [
      Object {
        "alignItems": "center",
        "borderRadius": 25,
        "height": 50,
        "justifyContent": "center",
        "marginRight": 15,
        "overflow": "hidden",
        "width": 50,
      },
      Object {
        "backgroundColor": "",
      },
    ]
  }
>
  <Image
    resizeMode="cover"
    source={Object {}}
    style={
      Object {
        "height": "100%",
        "width": "100%",
      }
    }
  />
</Component>
`;
